code/modules/mob/living/status_procs.dm

```diff
@@ -0,0 +1,513 @@
+	sleeping = max(amount, 0)
+	update_sleeping_effects()
+	if(updating)
+		update_stat()
+		update_canmove()
+
+/mob/living/AdjustSleeping(amount, bound_lower = 0, bound_upper = INFINITY, updating = 1)
+	var/new_value = directional_bounded_sum(sleeping, amount, bound_lower, bound_upper)
+	SetSleeping(new_value, updating)
+
+// SLOWED
+
+/mob/living/Slowed(amount, updating = 1)
+	SetSlowed(max(slowed, amount), updating)
+
+/mob/living/SetSlowed(amount, updating = 1)
+	slowed = max(amount, 0)
+
+/mob/living/AdjustSlowed(amount, bound_lower = 0, bound_upper = INFINITY, updating = 1)
+	var/new_value = directional_bounded_sum(slowed, amount, bound_lower, bound_upper)
+	SetSlowed(new_value, updating)
+
+// SLURRING
+
+/mob/living/Slur(amount)
+	SetSlur(max(slurring, amount))
+
+/mob/living/SetSlur(amount)
+	slurring = max(amount, 0)
+
+/mob/living/AdjustSlur(amount, bound_lower = 0, bound_upper = INFINITY)
+	var/new_value = directional_bounded_sum(slurring, amount, bound_lower, bound_upper)
+	SetSlur(new_value)
+
+// STUN
+
+/mob/living/Stun(amount, updating = 1, force = 0)
+	SetStunned(max(stunned, amount), updating, force)
+
+/mob/living/SetStunned(amount, updating = 1, force = 0) //if you REALLY need to set stun to a set amount without the whole "can't go below current stunned"
+	if(status_flags & CANSTUN || force)
+		stunned = max(amount, 0)
+		if(updating)
+			update_canmove()
+
+/mob/living/AdjustStunned(amount, bound_lower = 0, bound_upper = INFINITY, updating = 1, force = 0)
+	var/new_value = directional_bounded_sum(stunned, amount, bound_lower, bound_upper)
+	SetStunned(new_value, updating, force)
+
+// STUTTERING
+
+
+/mob/living/Stuttering(amount, force = 0)
+	SetStuttering(max(stuttering, amount), force)
+
+/mob/living/SetStuttering(amount, force = 0)
+	//From mob/living/apply_effect: "Stuttering is often associated with Stun"
+	if(status_flags & CANSTUN || force)
+		stuttering = max(stuttering, 0)
+
+/mob/living/AdjustStuttering(amount, bound_lower = 0, bound_upper = INFINITY, force = 0)
+	var/new_value = directional_bounded_sum(stuttering, amount, bound_lower, bound_upper)
+	SetStuttering(new_value, force)
+
+// WEAKEN
+
+/mob/living/Weaken(amount, updating = 1, force = 0)
+	SetWeakened(max(weakened, amount), updating, force)
+
+/mob/living/SetWeakened(amount, updating = 1, force = 0)
+	if(status_flags & CANWEAKEN || force)
+		weakened = max(amount, 0)
+		if(updating)
+			update_canmove()	//updates lying, canmove and icons
+
+/mob/living/AdjustWeakened(amount, bound_lower = 0, bound_upper = INFINITY, updating = 1, force = 0)
+	var/new_value = directional_bounded_sum(weakened, amount, bound_lower, bound_upper)
+	SetWeakened(new_value, updating, force)
+
+//
+//		DISABILITIES
+//
+
+// Blind
+
+/mob/living/proc/BecomeBlind()
+	var/val_change = !(disabilities & BLIND)
+	disabilities |= BLIND
+	if(val_change)
+		update_blind_effects()
+
+/mob/living/proc/CureBlind()
+	var/val_change = !!(disabilities & BLIND)
```

Comment:
It converts non-zero values to 1, but leaves 0 alone - not really needed, but leaves it as a definite "1 or 0" boolean, instead of an actual piece of information.